### PR TITLE
Validate build/deployment statuses and environments + update help

### PR DESCRIPTION
**Pull request by tovbinm**
Validate build/deployment statuses and environments + update help

### CODE REVIEW

File: faros_event.sh [hunk 1]
```diff
@@ -360,6 +367,15 @@ function resolveDeploymentInput() {
     deployment_status_details=${deploymant_status_details:-$FAROS_DEPLOYMENT_STATUS_DETAILS}
     deployment_start_time=${deployment_start_time:-$FAROS_DEPLOYMENT_START_TIME}
     deployment_end_time=${deployment_end_time:-$FAROS_DEPLOYMENT_END_TIME}
+
+    if ! [[ ${ENVS[*]} =~ (^|[[:space:]])"$deployment_env"($|[[:space:]]) ]] ; then
+      echo "Invalid deployment environment $deployment_env. Allowed values: ${envs}";
+      exit 1
```
Comment (willmarks) [inline faros_event.sh:373]: If you want I have those two helper functions `err` and `fail` which tack on the time and the level. For example:
```sh
if ! [[ ${ENVS[*]} =~ (^|[[:space:]])"$deployment_env"($|[[:space:]]) ]] ; then
      err "Invalid deployment environment $deployment_env. Allowed values: ${envs}";
      fail
fi
```
Which results in the following output:
Screen Shot 2021-07-16 at 8 18 04 PM

File: faros_event.sh [hunk 2]
```diff
@@ -59,23 +67,23 @@ function help() {
...
-    echo "--build <build>                       | FAROS_BUILD"
+    echo "--deployment_env <env>                | FAROS_DEPLOYMENT_ENV            | ${envs}"
+    echo "--deployment_status <status>          | FAROS_DEPLOYMENT_STATUS         | ${deployment_statuses}"
+    echo "--build <build>                       | FAROS_BUILD                     |"
     printf "${RED}(Required build fields)${NC}\\n"
-    echo "--build_status <status>               | FAROS_BUILD_STATUS"
-    echo "--repo <repo>                         | FAROS_REPO"
-    echo "--vcs_org <vcs_org>                   | FAROS_VCS_ORG"
-    echo "--vcs_source <vcs_source>             | FAROS_VCS_SOURCE"
+    echo "--build_status <status>               | FAROS_BUILD_STATUS              | ${build_statuses}"
```
Comment (willmarks) [inline faros_event.sh:83]: Nice! Good idea to add them here.

**Review by willmarks — APPROVED** [review]
Awesome! See comment around logging helpers, but looks good to me!

File: faros_event.sh (hunk 1 above)
Comment (tovbinm) [reply faros_event.sh]: Cool! Updated.

**Pull request merged**